forge/forge/verify/compare.py

```diff
@@ -0,0 +1,269 @@
+# SPDX-FileCopyrightText: © 2024 Tenstorrent AI ULC
+
+# SPDX-License-Identifier: Apache-2.0
+
+
+from typing import Union
+import os
+
```

Comment:
Is there a reason why you didn't send out the PR for resolving the circular dependency issue? If there is none, please send it, so that we can do this in chunks...